Examples/php/cpointer/runme.php

```diff
@@ -1,40 +1,38 @@
 <?php
 
-	require "example.php";
-
 	# First create some objects using the pointer library.
 
 	print "Testing the pointer library\n";
 
-	$a = example::new_intp();
-	$b = example::new_intp();
-	$c = example::new_intp();
-	example::intp_assign($a,37);
-	example::intp_assign($b,42);
+	$a = new_intp();
+	$b = new_intp();
+	$c = new_intp();
+	intp_assign($a,37);
+	intp_assign($b,42);
 
 	print "	a = $a\n";
 	print "	b = $b\n";
 	print "	c = $c\n";
 
 	# Call the add() function wuth some pointers
-	example::add($a,$b,$c);
+	add($a,$b,$c);
 
 	# Now get the result
-	$r = example::intp_value($c);
+	$r = intp_value($c);
 
 	print "	37 + 42 = $r\n";
 
 	# Clean up the pointers
-	example::delete_intp($a);
-	example::delete_intp($b);
-	example::delete_intp($c);
+	delete_intp($a);
+	delete_intp($b);
+	delete_intp($c);
 
 	# Now try the typemap library
 	# This should be much easier. Now how it is no longer
 	# necessary to manufacture pointers.
 
 	print "Trying the typemap library\n";
-	$r = example::sub(37,42);
+	$r = sub(37,42);
```

Comment:
It may work here, but `example::sub()` is returning an integer.  As I said above, the problems will be with objects, as in that case the flat functions (i.e. `x`) return a PHP resource wrapping the C++ object pointer while the class functions defined in the PHP wrapper (i.e. `example::x`) wrap a PHP object around the resource and return that instead.

> I didn't get what you mentioned about example class for compatibility.

`example::sub()` is the documented way this gets wrapped unless `-noproxy` is used.  If we stop wrapping things like that, then SWIG will suddenly generate different wrappers and all existing code relying on things like `example::sub()` working will stop working.  And that would annoy both users of SWIG and end users of wrappers generated with SWIG.

For example, users of Xapian's SWIG-generated PHP bindings are told to write `Xapian::DB_CREATE_OR_OPEN`, and a newer SWIG version which suddenly stopped generating a wrapper like that would be a problem.  Perhaps `DB_CREATE_OR_OPEN` can be safely used instead, but users of Xapian's PHP bindings will rightly be annoyed that they have to go and rewrite their scripts.  So with the new style of wrapping you're working on, I think we need to provide a way to still have this wrapping generated.

It seems I need to stress this again: we need to generate wrappers that are compatible with what's currently generated to the greatest extent possible.  If we don't, then it's not going to be reasonable to merge this work to git master, and it'll just sit on a branch.  There are already too many SWIG GSoC projects from previous years bitrotting on branches with decreasing hope of ever being merged as master naturally evolves and incompatibilities with the branch increase.  If we're going to approach project this with a disregard for compatibility, this will just be another dead project at the end of the summer and we might as well not bother spending the time working on it.